packages/flutter_tools/lib/src/ios/xcodeproj.dart

```diff
@@ -354,6 +356,23 @@ class XcodeProjectInterpreter {
   }
 }
 
+/// Environment variables prefixed by FLUTTER_XCODE_ will be passed as build configurations to xcodebuild.
+/// This allows developers to pass arbitrary build settings in without the tool needing to make a flag
+/// for or be aware of each one. This could be used to set code signing build settings in a CI
+/// environment without requiring settings changes in the Xcode project.
+List<String> environmentVariablesAsXcodeBuildSettings() {
+  const String xcodeBuildSettingPrefix = 'FLUTTER_XCODE_';
+  final Iterable<MapEntry<String, String>> xcodeEnvironmentVariables = platform.environment.entries.where((MapEntry<String, String> mapEntry) {
+    return mapEntry.key.startsWith(xcodeBuildSettingPrefix);
+  });
+  final Iterable<String> xcodeBuildSettingsKeyValue = xcodeEnvironmentVariables.expand<String>((MapEntry<String, String> mapEntry) {
```

Comment:
nit: I would chain all of these Iterable methods into a single expression. There is a lot of lhs real-estate dedicated to very temporary identifiers.

Reply:
@christopherfujino said:
https://github.com/flutter/flutter/pull/43553#discussion_r339814709

and then even claimed:
https://github.com/flutter/flutter/pull/43553#discussion_r339842589

Let me split the difference.